src/python/pants/backend/project_info/dependencies.py

```diff
@@ -47,22 +39,6 @@ def register_options(cls, register):
             default=False,
             help="Include the input targets in the output, along with the dependencies.",
         )
-        register(
-            "--type",
-            type=DependencyType,
-            default=DependencyType.SOURCE,
-            removal_version="2.9.0.dev0",
-            removal_hint="This option is misleading and not very useful. In the future there "
-            "will be a more robust way of querying and filtering dependencies.\nMeanwhile you "
-            "can get the list of requirement strings for a set of targets using something like\n\n"
-            "./pants dependencies :: \\\n"
```

Comment:
For the  suggested pipeline to work, which `_release_helper.py` uses in this review, the ambiguity in the deprecation is only solved by dependencies only printing what was `--type=source`; otherwise you get things like "foo=1.0.0" as input specs in the pipeline which causes failures.